### PR TITLE
Update the appliance documentation

### PR DESCRIPTION
**Pull request by settermjd**
This:

- removes any reference to "Enterprise"
- relates to https://github.com/owncloud/documentation/issues/3287

@michaelstingl can you comment, please?

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -55,12 +55,12 @@ This documents the *previous stable* release of ownCloud.
...
 ====================
-Enterprise Appliance
+ownCloud X Appliance
 ====================
 
-The ownCloud X Enterprise Appliance is a complete virtual machine image running ownCloud X Enterprise on Ubuntu Linux.
+The ownCloud X Appliance is a complete virtual machine image running ownCloud X, on Ubuntu Linux.
 
-* `ownCloud X Enterprise Appliance Manual <https://doc.owncloud.org/server/10.0/admin_manual/enterprise/appliance/index.html>`_
+* `ownCloud X Appliance Manual <https://doc.owncloud.org/server/10.0/admin_manual/enterprise/appliance/index.html>`_
```
Comment (michaelstingl) [inline index.rst]: why path `enterprise`? restructure?

**Review by michaelstingl — CHANGES_REQUESTED** [review]

File: index.rst [hunk 2]
```diff
@@ -55,12 +55,12 @@ This documents the *previous stable* release of ownCloud.
   <https://doc.owncloud.org/server/9.1/ownCloudDeveloperManual.pdf>`_)
 
 ====================
-Enterprise Appliance
+ownCloud X Appliance
 ====================
 
-The ownCloud X Enterprise Appliance is a complete virtual machine image running ownCloud X Enterprise on Ubuntu Linux.
+The ownCloud X Appliance is a complete virtual machine image running ownCloud X, on Ubuntu Linux.
```
Comment (michaelstingl) [inline index.rst]: Not Ubuntu, based on Univention Server that is based on Debian

**Review by michaelstingl — CHANGES_REQUESTED** [review]

**Comment by settermjd** [comment]
This was implemented in https://github.com/owncloud/documentation/pull/3296.

**Comment by settermjd** [comment]
@michaelstingl, can you review again, pls?

**Review by michaelstingl — APPROVED** [review]

**Pull request merged**